RelationalAI/DefaultAccessTokenHandler.cs

```diff
@@ -0,0 +1,106 @@
+﻿/*
+ * Copyright 2022 RelationalAI, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Runtime.InteropServices;
+using System.Threading.Tasks;
+using Newtonsoft.Json;
+
+namespace RelationalAI
+{
+    public class DefaultAccessTokenHandler : IAccessTokenHandler
+    {
+        public async Task<AccessToken> GetAccessTokenAsync(Rest rest, string host, ClientCredentials creds)
+        {
+            var token = ReadAccessToken(creds);
+            if (token != null && !token.IsExpired)
+            {
+                return token;
+            }
+
+            token = await rest.RequestAccessTokenAsync(host, creds);
```

Comment:
Given we await for the result, do it make sense to have `async Task<AccessToken>` as return type?

Reply:
Since `GetAccessTokenAsync` awaits for `RequestAccessTokenAsync` the return value is either a void or `Task<T>`

Reply:
ok it seems we have to mark `GetAccessTokenAsync` as `async` given we use `await` in its body :+1: